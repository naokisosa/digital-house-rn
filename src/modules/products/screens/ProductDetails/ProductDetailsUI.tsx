import { ScrollView, StyleSheet, View } from 'react-native';
import React from 'react';
import { BaseText, Button, Image, ViewContainer } from '@components';
import { colors } from '@style';
import { Product } from '@services/types';
import Header from '@modules/products/components/Header/Header';
import { formatDate } from '@utils';

interface Props {
  onPressBack: () => void;
  product?: Product;
}

const ProductDetailsUI = ({ onPressBack, product }: Props) => {
  return (
    <ViewContainer header={<Header title={product?.product} />} style={styles.container}>
      <ScrollView
        style={styles.content}
        showsVerticalScrollIndicator={false}
        alwaysBounceVertical={false}>
        <View style={[styles.card, styles.shadow]}>
          <Image size="big" source={{ uri: product?.image }} />
        </View>
        <BaseText style={styles.subtitle}>Detalles del producto:</BaseText>
        <BaseText style={styles.createdAt}>{`Comprado el ${formatDate(
          product?.createdAt
        )}`}</BaseText>
        <BaseText style={styles.subtitle}>Con esta compra acumulaste:</BaseText>
        <BaseText style={styles.points}>{`${product?.points} puntos`}</BaseText>
      </ScrollView>
      <Button title="Aceptar" onPress={onPressBack} />
    </ViewContainer>
  );
};

export default ProductDetailsUI;

const styles = StyleSheet.create({
  container: {
    paddingTop: 19,
  },
  content: {
    flex: 1,
  },
  subtitle: {
    fontSize: 14,
    fontWeight: '800',
    color: colors.gray,
    marginBottom: 19,
  },
  createdAt: {
    fontSize: 16,
    fontWeight: '800',
    marginBottom: 20,
  },
  points: {
    fontSize: 24,
    fontWeight: '800',
    marginTop: 13,
  },
  shadow: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.5,
    shadowRadius: 4,

    elevation: 2,
  },
  card: {
    backgroundColor: colors.white,
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
    height: 350,
    marginBottom: 32,
  },
});
