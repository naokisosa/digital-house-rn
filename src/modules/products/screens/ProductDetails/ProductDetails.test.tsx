import 'react-native';
import React from 'react';

import ProductDetailsUI from './ProductDetailsUI';
import renderer from 'react-test-renderer';
import { SINGLE_PRODUCT } from '@mocks';

describe('Test ProductDetails screen', () => {
  it('renders correctly with mock product', () => {
    const { toJSON } = renderer.create(
      <ProductDetailsUI product={SINGLE_PRODUCT} onPressBack={jest.fn} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
