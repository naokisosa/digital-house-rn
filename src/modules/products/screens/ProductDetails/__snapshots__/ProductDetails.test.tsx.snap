// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProductDetails screen renders correctly with mock product 1`] = `
[
  <View
    style={
      {
        "backgroundColor": "#CFD6FF",
        "height": 150,
        "justifyContent": "flex-end",
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Avenir",
          },
          {
            "fontSize": 24,
            "fontWeight": "800",
            "marginBottom": 24,
          },
        ]
      }
    >
      Recycled Wooden Salad
    </Text>
  </View>,
  <RNCSafeAreaView
    edges={
      [
        "bottom",
        "left",
        "right",
      ]
    }
    style={
      [
        {
          "backgroundColor": "#F8F8F8",
          "flex": 1,
          "paddingBottom": 6,
          "paddingHorizontal": 20,
          "paddingTop": 11,
        },
        {
          "paddingTop": 19,
        },
      ]
    }
  >
    <RCTScrollView
      alwaysBounceVertical={false}
      showsVerticalScrollIndicator={false}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 10,
                "height": 350,
                "justifyContent": "center",
                "marginBottom": 32,
              },
              {
                "elevation": 2,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.5,
                "shadowRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "overflow": "hidden",
                },
                [
                  {
                    "height": 200,
                    "width": 200,
                  },
                  undefined,
                ],
              ]
            }
          >
            <FastImageView
              defaultSource={
                {
                  "testUri": "../../../src/assets/placeholder.png",
                }
              }
              resizeMode="cover"
              source={
                {
                  "uri": "https://loremflickr.com/640/480/city",
                }
              }
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </View>
        </View>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "Avenir",
              },
              {
                "color": "#9B9898",
                "fontSize": 14,
                "fontWeight": "800",
                "marginBottom": 19,
              },
            ]
          }
        >
          Detalles del producto:
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "Avenir",
              },
              {
                "fontSize": 16,
                "fontWeight": "800",
                "marginBottom": 20,
              },
            ]
          }
        >
          Comprado el 9 de diciembre, 2022
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "Avenir",
              },
              {
                "color": "#9B9898",
                "fontSize": 14,
                "fontWeight": "800",
                "marginBottom": 19,
              },
            ]
          }
        >
          Con esta compra acumulaste:
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "Avenir",
              },
              {
                "fontSize": 24,
                "fontWeight": "800",
                "marginTop": 13,
              },
            ]
          }
        >
          44871 puntos
        </Text>
      </View>
    </RCTScrollView>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#334FFA",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "Avenir",
            },
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "800",
              "lineHeight": 21.86,
            },
          ]
        }
      >
        Aceptar
      </Text>
    </View>
  </RNCSafeAreaView>,
]
`;
